src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -320,29 +324,20 @@ public static bool GetKnownVersion(this ReadOnlySpan<byte> span, out HttpVersion
         /// The Known versions will be checked with the required '\r'.
         /// To optimize performance the HTTP/1.1 will be checked first.
         /// </remarks>
-        /// <returns><c>true</c> if the input matches a known string, <c>false</c> otherwise.</returns>
+        /// <returns>the HTTP version if the input matches a known string, <c>Unknown</c> otherwise.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static HttpVersion GetKnownVersionAndConfirmCR(this ReadOnlySpan<byte> location)
+        internal static HttpVersion GetKnownVersion(this ReadOnlySpan<byte> location)
         {
-            if (location.Length < sizeof(ulong))
+            Debug.Assert(location.Length > sizeof(ulong) && location[sizeof(ulong)] == (byte)'\r');
```

Comment:
This seems scary to me. Can we rename it to `GetKnownVersionUnchecked()` or something similar?

Reply:
I have renamed the mehtod to `GetKnownVersionUnchecked`.